### PR TITLE
Make VCSBeam GPU-agnostic

### PR DESCRIPTION
**Pull request by bwmeyers**
This PR contains a lot of the work done so far as part of the Pawsey Uptake project. We have successfully ported the code to work on Setonix (AMD GPUs) and Garrawarla (NVIDIA GPUs), the choice of which is made at compile time through a switch in the CMake recipe.

We are still in the process of optimising and searching for bugs/errors. So far, output beamformed data are identical. Expect further updates, but this is a major milestone and will constitute a new major version and corresponding release.

### CODE REVIEW

**Comment by bwmeyers** [comment]
Some clean-up of the source files and headers might still be necessary, but for the moment everything is reasonably laid out and compiles.

File: app/make_mwa_incoh_beam.c
Comment (robotopia) [inline app/make_mwa_incoh_beam.c:338]: We just had a discussion in STDERR on Monday about how `char` can actually be interpreted as `unsigned char` or `signed char` depending on the target architecture and compiler. To avoid potential errors later on different systems, the general advice is to always explicitly say whether you want `signed` or `unsigned`.

I won't let this hold up the PR, but I'll let you (whoever is reading this) decide whether this should be made into an "issue" to be followed up later, in order to make this code as portable as possible.

**Review by robotopia — APPROVED** [review]

File: app/make_mwa_incoh_beam.c
Comment (bwmeyers) [reply app/make_mwa_incoh_beam.c:338]: Cool. I have made it an issue, but like you say I won't let it hold this up.

**Pull request merged**
